### PR TITLE
feat!: monorepo

### PR DESCRIPTION
**Pull request by SrIzan10**
# This PR has been WIP for about 4 days

hello! the elephant in the room on this PR is moving to a monorepo structure.

Apart from this, I also made other stuff to start off on the right foot this 2024:

- init next.js frontend
- basic authentication done
- move to postgres + drizzle orm (see database package)
- sharing schema between frontend and backend
... with more to come!

PD: thanks everyone for the support

### CODE REVIEW

**Comment by EvolutionX-10** [comment]
If this is not urgent, i'd review it in the first week of Jan as my end semester exams are going on right now, otherwise @jacoobes and @Murtatrxx can review it on my behalf as well

**Comment by SrIzan10** [comment]
alright, if both of them review before you can it’s fine
gl!

File: apps/api/util/setup.sh [hunk 1]
```diff
@@ -38,11 +39,6 @@ echo -ne "Installing yarn"
 npm install -g yarn > /dev/null 2>&1
 echo " done"
 
-echo -ne "Setting babashka preloads"
-BABASHKA_PRELOADS='(def lines (babashka.fs/read-all-lines "../.env"))'
-BABASHKA_PRELOADS=$BABASHKA_PRELOADS' (def env (into (sorted-map) ( map #(str/split % #"=") lines )))'
```
Comment (jacoobes) [inline apps/api/util/setup.sh:43]: we'll have to fix this in the future. Might be able to move to nbb which has better support with nodejs

Comment (SrIzan10) [reply apps/api/util/setup.sh:43]: yes that's why i removed it

File: apps/frontend/src/server/db/authAdapter.ts [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+      await db
+        .delete(accounts)
+        .where(
+          and(
+            eq(accounts.providerAccountId, account.providerAccountId),
+            eq(accounts.provider, account.provider)
+          )
+        );
+
+      return undefined;
```
Comment (jacoobes) [inline apps/frontend/src/server/db/authAdapter.ts]: no use in returning anything here

File: apps/frontend/src/server/db/authAdapter.ts [hunk 3]
```diff
@@ -0,0 +1,144 @@
+// @ts-nocheck shush lets pretend this is fine
+import { accounts, sessions, users, verificationTokens } from "database/src/schema";
+import type { PostgresJsDatabase } from "drizzle-orm/postgres-js";
+import { and, eq } from "drizzle-orm";
+import type { Adapter } from "next-auth/adapters";
+
+export function PostgresJsDrizzleAdapter(db: PostgresJsDatabase): Adapter {
+  return {
+    createUser: async (data) => {
+      return await db
```
Comment (jacoobes) [inline apps/frontend/src/server/db/authAdapter.ts]: return await ❌

**Review by jacoobes — CHANGES_REQUESTED** [review]

File: apps/frontend/src/server/db/authAdapter.ts (hunk 2 above)
Comment (SrIzan10) [reply apps/frontend/src/server/db/authAdapter.ts]: you mean just `return`ing?

**Comment by SrIzan10** [comment]
hey @EvolutionX-10 I'm removing you as reviewer. good luck with the exams!

**Review by jacoobes — APPROVED** [review]

**Pull request merged**